activesupport/lib/active_support/dependencies.rb

```diff
@@ -179,6 +179,19 @@ def pop_modules(modules)
     mattr_accessor :constant_watch_stack
     self.constant_watch_stack = WatchStack.new
 
+    # Module prepends this
+    module ModuleConstGet
+      def const_get(const_name, *)
+        if !anonymous? && self != Object
```

Comment:
A potential style tweak here to minimize the amount of `!` logic to read through. If I'm doing the boolean logic right in my head, I think you can do something like 
```ruby
unless anonymous? || self == object
  ...
ensure
  ...
end
```